src/oc/fileio.cpp

```diff
@@ -947,4 +947,22 @@ int Fprintf(FILE* stream, const char *fmt, ...) {
   return n;
 }
 
+void hoc_execerr_ext(const char* fmt, ...) {
+  size_t size;
+  va_list ap;
+
+  va_start(ap, fmt);
+  size = vsnprintf(NULL, 0, fmt, ap);
+  va_end(ap);
+  if (size >= 0) {
+    char s[size + 1];
```

Comment:
nitpick : I hope size is not too large that will cause errors due to stack allocation

Reply:
I wondered about that.  How would one know what the available size is?  Is there a better way to avoid a memory leak since the following hoc_execerror will not return.  I guess I just concluded that error messages are generally not very long, but I suppose I could add a test to demand size < some value.

Reply:
> I wondered about that. How would one know what the available size is? I

Haven't used but not sure if alloca is better in any way : https://www.man7.org/linux/man-pages/man3/alloca.3.html

Reply:
I looked into this further and I gather that although runtime allocation of arrays on the stack work for g++ and clang, they are not a part of the c++ standard. I'll change to ```std::vector```  (note that as is, there is no -Wall warning for this).

Reply:
Available stack size is os-dependent, not sure there is a cross-platform way of retrieving it. The best way to avoid error leaks is to use `std::vector` as you suggested.

Reply:
Ok. hoc_execerr_ext(const char* fmt, ...) truncates message after 512 bytes. No memory leak.